src/errors-new.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  MatcherLocation,
+  MatcherLocationNormalized,
+  RouteLocation,
+  RouteLocationNormalized,
+} from './types'
+
+// Using string enums because error codes are exposed to developers
+// and number enums could collide with other error codes in runtime
+export enum ErrorTypes {
+  NO_ROUTE_MATCH = 'NO_ROUTE_MATCH',
```

Comment:
Let's put `MATCHER_` in front
```suggestion
  MATCHER_NOT_FOUND = 'MATCHER_NOT_FOUND',
```